src/Compilers/CSharp/Portable/Symbols/MethodGroupConversionCaches/MethodGroupConversionCacheFrame.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// This class holds ALL types of delegates that are coverted from ONE target method
+    /// Type parameters are generated for: <see cref="GetTypeArgumentsFromConversion(TypeSymbol, MethodSymbol)"/>
+    ///     1. All type parameters from the containing types of the target method
+    ///     2. All type parameters from the target method
+    ///     3. The delegate type
+    /// </summary>
+    /// <remarks>
+    /// This symbol is created in lowering phrase, and is collected and translated in emit phrase.
+    /// </remarks>
+    internal sealed class MethodGroupConversionCacheFrame : NamedTypeSymbol
```

Comment:
I don't think we should model it based upon neither anonymous types nor lambdas. 

Anonymous types add extra complexity since they have to be unified and thus assigned late in the emit phase. 

Lambdas add complexity since thy have to be reused during EnC. Method group caches don't have to be reused. 

I'd suggest using similar mechanism to *dynamic* call-site containers.